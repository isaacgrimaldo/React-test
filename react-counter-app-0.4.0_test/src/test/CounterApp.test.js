import React from 'react';
import'@testing-library/jest-dom';
import  {shallow}  from 'enzyme';

import CounterApp from './../CounterApp';

describe('Prueva del archivo CounterApp.js', () =>{
    let wrapper = shallow( <CounterApp/> ); // Variables globales;
    
    beforeEach( () => { // Se ejecuta antes de cada prueba 
          wrapper = shallow( <CounterApp/> );
    })
   

    test('<CounterApp/> Debe renderizarce correctamente el HTML con los valores por defecto', () => {
       expect(wrapper).toMatchSnapshot();
     });

     test('<CounterApp/> Debe mostrar correctamente la propiedad "value"', () => {
         const testValue = 100;
         const wrapper = shallow( <CounterApp value={testValue}/> );
         const  counter = wrapper.find('h2').text().trim();
         expect(counter).toBe('100');
     });

     test('Debe de funcionar el boton +1', () =>{
        wrapper.find('button').at(0).simulate('click');
        const  counter = wrapper.find('h2').text().trim();
        expect(counter).toBe('11');
     });

     test('Debe de funcionar el boton -1', () =>{
        wrapper.find('button').at(2).simulate('click');
        const  counter = wrapper.find('h2').text().trim();
        expect(counter).toBe('9');
     });

    test('Debe de funcionar el botÃ³n rest', () =>{
        const wrapper = shallow( <CounterApp value={150}/> );
        
        for(let i=0; i < 10; i++  ){
            wrapper.find('button').at(0).simulate('click');
        };
        for(let i=0; i < 5; i++  ){
            wrapper.find('button').at(2).simulate('click');
        };

        wrapper.find('button').at(1).simulate('click');

        const counter = wrapper.find('h2').text().trim();
 ;      expect(counter).toBe('150')
    })

;
})