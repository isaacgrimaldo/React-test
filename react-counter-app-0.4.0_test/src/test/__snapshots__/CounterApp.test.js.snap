// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueva del archivo CounterApp.js <CounterApp/> Debe renderizarce correctamente el HTML con los valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
